py-polars/tests/unit/interop/numpy/test_ufunc_expr.py

```diff
@@ -130,3 +131,24 @@ def test_ufunc_multiple_expressions() -> None:
 def test_grouped_ufunc() -> None:
     df = pl.DataFrame({"id": ["a", "a", "b", "b"], "values": [0.1, 0.1, -0.1, -0.1]})
     df.group_by("id").agg(pl.col("values").log1p().sum().pipe(np.expm1))
+
+
+@guvectorize([(float64[:], float64[:])], "(n)->(n)")
+def gufunc_mean(arr, result):  # type: ignore[no-untyped-def]
+    mean = arr.mean()
+    for i in range(len(arr)):
+        result[i] = mean + i
+
+
+def test_generalized_ufunc() -> None:
```

Comment:
It's not really testing Expr.__array_ufunc__ if you use map_batches. I was thinking tests like `df.select(gufunc_mean(pl.col('s')).alias('result')` That said, I don't think you need to put those tests in this PR as it's only tangentially related.

Reply:
OK.